packages/backend-plugin-api/src/services/definitions/RootLifecycleService.ts

```diff
@@ -23,4 +23,6 @@ import { LifecycleService } from './LifecycleService';
  *
  * @public
  */
-export interface RootLifecycleService extends LifecycleService {}
+export interface RootLifecycleService extends LifecycleService {
+  addPreShutdownHook(hook: () => void): void;
```

Comment:
Naming wise just wanna float some options, these are all for the pattern `add...Hook`:

```
PreShutdown
PrepareShutdown
Drain
TrafficDrain
ConnectionDrain
Close
BeforeShutdown
```

Reply:
Based on discussion on Discord we're moving forward with `addBeforeShutdownHook`

Reply:
Name updated 🙌🏻